client/src/app/home/feature/home/home.page.ts

```diff
@@ -87,6 +91,15 @@ export class HomePage {
     });
   }
 
+  filterUsers(event: Event) {
+    const searchQuery = (event.target as HTMLInputElement).value.toLowerCase();
+    if (searchQuery === '') {
+      this.filteredUsersWithRating = this.usersWithRating;
+    } else {
+      this.filteredUsersWithRating = this.usersWithRating.filter(user => user.name.toLowerCase().includes(searchQuery));
```

Comment:
```suggestion
      this.filteredUsersWithRating = this.usersWithRating.filter(user => user.name.toLowerCase().includes(searchQuery)) || [];
```
Add fallback case for when it searchs for when the query doesn't match any username.